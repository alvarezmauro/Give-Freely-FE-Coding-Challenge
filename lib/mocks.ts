import type { participantsDataType } from "@/types/participantsData";

// Data we expect to be stored in storage
export const MOCK_STORED_PARTICIPANTS_DATA: participantsDataType = {
  websites: [
    {
      name: "TripAdvisor",
      url: "www.tripadvisor.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    }
  ]
};

// Data we expect to be returned from getParticipantsData
export const MOCK_PARTICIPANTS_DATA: participantsDataType = {
  websites: [
    {
      name: "TripAdvisor",
      url: "www.tripadvisor.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    },
    {
      name: "Uber",
      url: "www.uber.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    },
    {
      name: "Amazon",
      url: "www.amazon.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    }
  ]
};
