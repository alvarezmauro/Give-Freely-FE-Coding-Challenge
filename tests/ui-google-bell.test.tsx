import GoogleBell from "@/contents/ui-google-bell";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import { MOCK_PARTICIPANTS_DATA } from "./mocks/dataMocks";

// Mock @plasmohq/storage/hook
jest.mock("@plasmohq/storage/hook", () => ({
  useStorage: () => [MOCK_PARTICIPANTS_DATA]
}));

// mock import "@/style.css";
jest.mock("@/style.css", () => ({}));

describe("ui-google-bell", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Snapshot test of bell button", async () => {
    render(<GoogleBell />);
    const renderedGoogleBellButton = await screen.queryByTestId(
      "google-bell-button"
    );
    expect(renderedGoogleBellButton).toMatchSnapshot();
  });

  it("Snapshot test of bell button", async () => {
    render(<GoogleBell />);
    const renderedGoogleBellButton = await screen.queryByTestId(
      "google-bell-button"
    );
    // Click on the bell button
    await act(async () => {
      // Click on the bell button
      await renderedGoogleBellButton.click();
    });
    // Check if the dialog is rendered
    const renderedGoogleBellDialog = await screen.queryByTestId(
      "google-bell-dialog"
    );
    expect(renderedGoogleBellDialog).toBeInTheDocument();
  });
});
