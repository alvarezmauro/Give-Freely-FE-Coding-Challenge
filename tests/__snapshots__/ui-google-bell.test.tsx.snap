// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-google-bell Snapshot test of bell button 1`] = `
<button
  aria-controls="radix-:r0:"
  aria-expanded="false"
  aria-haspopup="dialog"
  data-state="closed"
  data-testid="google-bell-button"
  style="position: fixed; right: 1rem; bottom: 1rem;"
  type="button"
>
  <svg
    class="mr-2 h-4 w-4"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 8a6 6 0 0 1 12 0c0 7 3 9 3 9H3s3-2 3-9"
    />
    <path
      d="M10.3 21a1.94 1.94 0 0 0 3.4 0"
    />
  </svg>
</button>
`;
